src/contexts/EthereumContext.tsx

```diff
@@ -207,13 +222,62 @@ export function EthereumProvider({ children }: { children: React.ReactNode }) {
   // Disconnect the provider by resetting the app's account
   const disconnect = () => {
     setAccount(null)
+    setVerified(false)
   }
 
   // Toggle function
   const toggleOnboard = () => {
     setUseOnboard(!useOnboard)
   }
 
+  // Sign In With Ethereum
+  const signInWithEthereum = async () => {
+    // SIWE requires an connected account
+    if (!account || !provider) {
+      console.log('No account connected')
+      return
+    }
+
+    // Get chain ID of provider.
+    const { chainId } = await provider.getNetwork()
+
+    const siweMessage = new SiweMessage({
+      domain: window.location.host,
+      address: account,
+      statement: 'By logging in you agree to the terms and conditions.',
+      uri: window.location.origin,
+      version: '1',
+      chainId: Number(chainId),
+      resources: ['https://boilerplate.lukso.tech'],
+    })
+
+    const message = siweMessage.prepareMessage()
+    console.log('firstmessage:', message)
+    const hashedMessage = ethers.hashMessage(message)
+
+    try {
```

Comment:
Why would you put such a complex function in a context?

Imo this function does not need to be in any context, just an export somewhere

Reply:
Thanks. I was considering putting it as an external function too, but first kept it inside because the `isVerified` property is carried within. But I agree, It is better if it's scope.

Implemented it as an external function 💪🏻 ✅